Platform/ElkhartlakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c

```diff
@@ -139,7 +178,7 @@ GetBoardIdFromSmbus (
     DEBUG ((DEBUG_INFO, "Fields.SpdPresent from Smbus Io expander is %x\n", (BOOLEAN) (SmbusInputPort0Info.InputPort0Fields.SpdPresent)));
   } else {
     DEBUG ((DEBUG_ERROR, "Failed to get Board ID from Smbus Io expander\n"));
-    *BoardId = BoardIdEhlLp4xType3Crb;
+    GetBoardIdFromUP2 (BoardId);
```

Comment:
I think it might have issue here.   Do we know other board won't have those GPIO set to 0 ?
so you might get a wrong board detection ? 
It might be better to follow all other AAEON board support to specific board ID during the stitching process.  So no board detection is really required.